lighthouse-core/lib/network-request.js

```diff
@@ -341,10 +370,101 @@ module.exports = class NetworkRequest {
+    const floatValue = parseFloat(totalFetchedSize.value);
+    // Bail if the header cannot be parsed.
+    if (isNaN(floatValue)) return;
+    this.transferSize = floatValue;
+  }
+
+  /**
+   * LR gets additional, accurate timing information from its underlying fetch infrastructure.  This
+   * is passed in via X-Headers similar to 'X-TotalFetchedSize'.
+   *
+   * TODO(exterkamp): Uncomment breaking code and apply these timings.
+   * see: https://github.com/GoogleChrome/lighthouse/issues/7752
+   */
+  _updateTimingsForLightrider() {
+    // Bail if we aren't in Lightrider.
+    if (!global.isLightrider) return;
+
+    // Bail if timing is not initialized.
+    if (!this.timing) {
+      return;
+    }
+
+    // For more info on timing nomenclature: https://www.w3.org/TR/resource-timing-2/#processing-model
+
+    //    StartTime
+    //    | ConnectStart
+    //    | |     SSLStart  SSLEnd
+    //    | |     |         | ConnectEnd
+    //    | |     |         | | SendStart/End   ReceiveHeadersEnd
+    //    | |     |         | | |               |                EndTime
+    //    ▼ ▼     ▼         ▼ ▼ ▼               ▼                ▼
+    //    [ [TCP  [   SSL   ] ] [   Request   ] [   Response   ] ]
+    //    ▲ ▲     ▲         ▲ ▲ ▲             ▲ ▲              ▲ ▲
+    //    | |     '-SSLMs---' | '-requestMs---' '-responseMs---' |
+    //    | '----TCPMs--------'                                  |
+    //    |                                                      |
+    //    '------------------------TotalMs-----------------------'
+
+    const totalHeader = this.responseHeaders.find(item => item.name === HEADER_TOTAL);
+    // Bail if there was no totalTime.
+    if (!totalHeader) return;
+
+    const totalMs = parseInt(totalHeader.value);
+
+    const TCPMsHeader = this.responseHeaders.find(item => item.name === HEADER_TCP);
+    const requestMsHeader = this.responseHeaders.find(item => item.name === HEADER_REQ);
+    const SSLMsHeader = this.responseHeaders.find(item => item.name === HEADER_SSL);
+    const responseMsHeader = this.responseHeaders.find(item => item.name === HEADER_RES);
+
+    // TODO(exterkamp): NaN checking
+    // Make sure all Times are initialized and are non-negative.
+    const TCPMs = TCPMsHeader ? Math.max(0, parseInt(TCPMsHeader.value)) : 0;
+    const requestMs = requestMsHeader ? Math.max(0, parseInt(requestMsHeader.value)) : 0;
+    const _/** SSLMs */ = SSLMsHeader ? Math.max(0, parseInt(SSLMsHeader.value)) : 0;
+    const responseMs = responseMsHeader ? Math.max(0, parseInt(responseMsHeader.value)) : 0;
+
+    // Bail if the timings don't add up.
+    if (TCPMs + requestMs + responseMs !== totalMs) {
+      return;
+    }
+
+    /*
+    This code will change performance scores.  The current Lighthouse version is 4.x,
+    and does not take into account these changes.  In order to uncomment and apply these
+    changes the main version number will need to be bumped to 5.x+.
```

Comment:
please delete the commented out stuff and put it in a new PR blocked on 5.0. No need to ship commented-out future code :)

Reply:
this is true.

Reply:
Done!